proto/api.proto

```diff
@@ -8,10 +8,20 @@ import public "info.proto"; // statistics about the request, filled out by loki/
 import public "status.proto"; // info for status endpoint
 
 message Api {
-  optional Options options = 1;
-  optional Trip trip = 2;
-  optional Directions directions = 3;
-  optional Info info = 4;
-  optional Status status = 5;
+  oneof has_options {
+    Options options = 1;
+  }
+  oneof has_trip {
+    Trip trip = 2;
+  }
+  oneof has_directions {
+    Directions directions = 3;
+  }
+  oneof has_info {
+    Info info = 4;
+  }
+  oneof has_status {
+    Status status = 5;
+  }
```

Comment:
actually none of these needed to be `oneof` we only need it for scalar fields. ill revert this